tests/tests_integration/test_api/test_assets.py

```diff
@@ -101,14 +101,26 @@ def test_get_subtree(self, root_test_asset):
         assert 6 == len(COGNITE_CLIENT.assets.retrieve_subtree(root_test_asset.id, depth=1))
 
     def test_create_asset_hierarchy_parent_external_id_not_in_request(self):
-        root_external_id = "my_root_{}".format(utils._auxiliary.random_string(5))
+        root_external_id = "my_root_{}".format(utils._auxiliary.random_string(10))
         root = Asset(external_id=root_external_id, name="my_root")
-        root = COGNITE_CLIENT.assets.create(root)
-
         children = generate_asset_tree(
             root_external_id=root_external_id, depth=5, children_per_node=10, current_depth=2
         )
-        children = COGNITE_CLIENT.assets.create(children)
-        assert COGNITE_CLIENT.assets._CREATE_LIMIT < len(children)
 
-        COGNITE_CLIENT.assets.delete(external_id=root_external_id, recursive=True)
+        try:
+            COGNITE_CLIENT.assets.create(root)
+            COGNITE_CLIENT.assets.create(children)
+
```

Comment:
The setup could be put in a fixture, but since the main purpose of the test is to test the create() functions, I felt it was more natural to include it in the test. What do you think?

Reply:
Up to you really, putting it in a fixture would make it more similar to the above test, so maybe easier to understand. But I agree with your reasoning.